libs/test/SalesforceAnalyticsTest/src/com/salesforce/androidsdk/analytics/security/EncrypterStreamTest.java

```diff
@@ -0,0 +1,195 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.androidsdk.analytics.security;
+
+import android.content.Context;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import java.io.BufferedReader;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test for {@link EncrypterOutputStream} and {@link DecrypterInputStream}
+ *
+ */
+@RunWith(AndroidJUnit4.class)
+public class EncrypterStreamTest {
+
+    private static final String TEST_FILE = "encrypter_stream_test_file";
+
+    private Context context;
+    private String encryptionKey;
+
+    @Before
+    public void setUp() throws Exception {
+        encryptionKey = Encryptor.hash("test-key", "hashing-key");
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        context.deleteFile(TEST_FILE);
+    }
+
+    /**
+     * Write an encrypted file using EncrypterOutputStream and read it back using
+     * DecrypterInputStream
+     */
+    @Test
+    public void testWriteAndReadThroughStream() {
+        String contentToWrite = "testWriteAndReadThroughStream";
+        writeThroughStream(contentToWrite);
+        String readContent = readThroughStream();
+        Assert.assertEquals(contentToWrite, readContent);
+    }
+
+    /**
+     * Write an encrypted file using Encryptor.encrypt and a FileOutputStream and read it back using
+     * DecrypterInputStream
+     *
+     * <p>Note: won't work because Encryptor.encrypt base-64 encodes content
+     */
+    // @Test
+    public void testWriteWithEncryptorAndReadThroughStream() {
+        String contentToWrite = "testWriteWithEncryptorAndReadThroughStream";
+        writeWithEncryptor(contentToWrite);
+        String readContent = readThroughStream();
+        Assert.assertEquals(contentToWrite, readContent);
+    }
+
+    /**
+     * Test that writes an encrypted file using EncrypterOutputStream and reads it back using
+     * FileInputStream and Encryptor.decrypt
+     *
+     * <p>Note: won't work because Encryptor.decrypt expect the encrypted content base 64 encoded
+     */
+    // @Test
+    public void testWriteThroughStreamAndReadWithEncryptor() {
```

Comment:
Can't write with Encryptor and read with the new stream classes.
Keeping that test around until I add the code to facilitate the migration (maybe as part of this PR).